core_lib_generator/core_lib_template_generator.py

```diff
@@ -0,0 +1,235 @@
+            },
+        }
+    return db_template
+
+
+def generate_solr_template() -> dict:
+    solr_protocol = input_str('Enter solr protocol (http/https)', 'https')
+    solr_port = input_int('Enter solr port no.', 8983)
+    solr_host = input_str('Enter solr host', 'localhost')
+    solr_file = input_str('Enter solr file name (solr/core_name)', 'solr/mycore')
+    print('\nInput recorded.')
+    return {
+        'env': {
+            'SOLR_HOST': solr_host,
+            'SOLR_PORT': solr_port,
+        },
+        'config': {
+            'protocol': solr_protocol,
+            'host': f'${{oc.env:SOLR_HOST}}',
+            'port': f'${{oc.env:SOLR_PORT}}',
+            'file': solr_file,
+        },
+    }
+
+
+def generate_cache_template() -> dict:
+    cache_type = input_enum(CacheTypes, 'for cache type')
+
+    if cache_type == 1:
+        memcache_port = input_int('Enter your memcached server port no.', 11211)
+        memcache_host = input_str('Enter your memcached server host', 'localhost')
+        print(f'\nCache type {CacheTypes(cache_type).name} on {memcache_host}:{memcache_port}')
+        return {
+            'env': {
+                'MEMCACHED_HOST': memcache_host,
+                'MEMCACHED_PORT': memcache_port,
+            },
+            'config': {
+                'type': CacheTypes(cache_type).name.lower(),
+                'host': f'${{oc.env:MEMCACHED_HOST}}',
+                'port': f'${{oc.env:MEMCACHED_PORT}}',
+            },
+        }
+    elif cache_type == 2:
+        print(f'\nCache type {CacheTypes(cache_type).name}')
+        return {
+            'config': {
+                'type': CacheTypes(cache_type).name.lower(),
+            }
+        }
+    else:
+        print('\nNo cache set.')
+
+
+def generate_db_entity_template() -> dict:
+    entities = {}
+    add_entity = True
+    while add_entity:
+        is_soft_delete = False
+        is_soft_delete_token = False
+        entity_name = input_str('Enter the name of the entity you\'d like to create: ', 'User')
+        while entity_name in entities:
+            entity_name = input_str('Enter the name of the entity you\'d like to create: ', 'User')
+        column_count = input_int('How many columns will you have in your entity? ', 0)
+        columns = {}
+        if column_count != 0:
+            for i in range(0, column_count):
+                column_name = input_str(f'Enter the name of column #{i + 1}: ')
+
+                column_type = input_enum(DBDatatypes, f'for datatype #{i + 1}')
+
+                column_default = input_str(f'Enter the default value of column #{i + 1}', ' ')
+
+                columns[column_name] = {
+                    'name': column_name,
+                    'type': DBDatatypes(column_type).name,
+                    'default': column_default
+                }
+
+            is_soft_delete = input_yes_no('Do you want to implement Soft Delete?', False)
+            if is_soft_delete:

```

Comment:
no duplicate logic

Reply:
this is for asking if `Soft delete` is not selected by user then ` we should not ask for `Soft delete token` right?